File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)

    for i := 0; i < len(nums); i++{
        complement := target - nums[i]

        if index, ok := m[complement]; ok {
            return []int{index, i}
        }

        m[nums[i]] = i;
    }

    return []int{}
}

File: 1-two-sum/two-sum.cpp
class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        vector<int> ans(2);

        unordered_map<int, int> mp;

        for(int i = 0; i < nums.size(); i++)
        {
            int comp = target - nums[i];

            auto it = mp.find(comp);

            if(it != mp.end()) {
                ans[0] = it->second;
                ans[1] = i;
            }
            mp[nums[i]] = i;
        }
        return ans;
    }
};